### PR TITLE
3243 remove gamfun and replace it with the analytical solution from scipy

### PR DESCRIPTION
**Pull request by chris-ashe**
## Description

The use of `gamfun()` from `maths_library.f90` within `plasma_profiles.py` for the temperature profile has been removed and replaced with `scipy.special.gamma`

## Checklist

I confirm that I have completed the following checks:

- [x] I have justified any large differences in the regression tests caused by this pull request in the comments.
- [x] I have added new tests where appropriate for the changes I have made.
- [x] If I have had to change any existing unit or integration tests, I have justified this change in the pull request comments.
- [x] If I have made documentation changes, I have checked they render correctly.
- [x] I have added documentation for my change, if appropriate.

### CODE REVIEW

**Review by j-a-foster — COMMENTED** [review]
All looks good to me. I'm assuming where you've removed `maths_library` that it's only imported to call `gamfun`?

**Comment by chris-ashe** [comment]
@j-a-foster  Have reverted the commit as it was causing problems with `init.f90` and their is still some function used from `maths_functions.f90`

File: tests/unit/test_plasma_profiles.py [hunk 1]
```diff
@@ -246,15 +246,15 @@ class PlasmaProfilesParam(NamedTuple):
...
-            expected_ten=14.521871327399182,
-            expected_tin=14.521871327399182,
+            expected_ten=14.52233022043558,
+            expected_tin=14.52233022043558,
             expected_alphap=2.4500000000000002,
-            expected_te0=27.369013322953624,
-            expected_p0=868071.46874220832,
-            expected_pcoef=1.1110842637642833,
+            expected_te0=27.370104119511087,
+            expected_p0=868106.0658743214,
```
Comment (timothy-nunn) [inline tests/unit/test_plasma_profiles.py:253]: This change is quite big. Any idea?

File: tests/unit/test_plasma_profiles.py [hunk 2]
```diff
@@ -291,15 +291,15 @@ class PlasmaProfilesParam(NamedTuple):
...
```
Comment (timothy-nunn) [inline tests/unit/test_plasma_profiles.py:298]: Again, why so much bigger than the others

**Review by timothy-nunn — CHANGES_REQUESTED** [review]

File: tests/unit/test_plasma_profiles.py (hunk 1 above)
Comment (chris-ashe) [reply tests/unit/test_plasma_profiles.py:253]: The calculation is several gamma terms multiplied together so the divergence in result is compounded at the output

File: tests/unit/test_plasma_profiles.py (hunk 2 above)
Comment (chris-ashe) [reply tests/unit/test_plasma_profiles.py:298]: ^ see above

**Review by timothy-nunn — CHANGES_REQUESTED** [review]
Line 52 (`integrate.simpson`) needs to become `sp.integrate...`

**Pull request merged**